solr/core/src/java/org/apache/solr/search/facet/FacetRangeProcessor.java

```diff
@@ -774,6 +763,30 @@ public Long parseAndAddGap(@SuppressWarnings("rawtypes") Comparable value, Strin
     }
   }
 
+  private static class EnumCalc extends Calc {
+
+    private final EnumMapping mapping;
+    public EnumCalc(final SchemaField f) {
+      super(f);
+      mapping = ((AbstractEnumField)field.getType()).getEnumMapping();
+    }
+    @Override
+    public EnumFieldValue bitsToValue(long bits) {
+      Integer val = (int)bits;
+      return new EnumFieldValue(val, mapping.intValueToStringValue(val));
+    }
+    @Override
+    protected EnumFieldValue parseStr(String rawval) {
+      Integer val = Integer.valueOf(rawval);
+      return new EnumFieldValue(val, mapping.intValueToStringValue(val));
+    }
+    @Override
+    protected EnumFieldValue parseAndAddGap(@SuppressWarnings("rawtypes") Comparable value, String gap) {
```

Comment:
The test didn't exercise this.  The impl looks suspicious because it parses a String to an integer.